src/services/platforms-service.ts

```diff
@@ -0,0 +1,46 @@
+import { Subject } from "joyst";
+
+export type Platform = {
+    x: number;
+    y: number;
+    length: number;
+    spawnPointsCount: number;
+    id: number;
+};
+
+export class PlatformsService {
+    public static platforms = new Subject<Subject<Platform>[]>([]);
+
+    private static nextId = 0;
+
+    public static add(): Subject<Platform> {
+        const newId = this.nextId++;
+
+        const newPlatformSubject = new Subject({
+            x: 0,
+            y: 0,
+            length: 1,
+            spawnPointsCount: 0,
+            id: newId
+        }, `platform-${newId}`);
+
+        this.platforms.set([
+            ...this.platforms.get(),
+            newPlatformSubject
+        ]);
+
+        return newPlatformSubject;
+    }
+
+    public static remove(subject?: Subject<Platform>): void {
+        if (!subject) {
+            return;
+        }
```

Comment:
What's the case where `subject` might be undefined?

Reply:
`remove` is called with the value provided by a custom event's detail property and in theory you could provide a value there that does not resolve to a subject so when you pass it to remove here the subject would be undefined.